R/derive_vars_dt.R

```diff
@@ -443,42 +428,40 @@ impute_dtc_dt <- function(dtc,
     month = "M",
     year = "Y"
   )
-  assert_character_scalar(highest_imputation, values = imputation_levels)
+
+  assert_highest_imputation(
+    highest_imputation = highest_imputation,
+    highest_imputation_values = imputation_levels,
+    date_imputation = date_imputation,
+    min_dates = min_dates,
+    max_dates = max_dates
+  )
+
   highest_imputation <- dt_level(highest_imputation)
-  date_imputation <-
-    assert_character_scalar(
-      date_imputation,
-      case_sensitive = FALSE
-    )
+
   assert_logical_scalar(preserve)
 
-  # Parse character date ----
-  two <- "(\\d{2}|-?)"
-  partialdate <- str_match(dtc, paste0(
-    "(\\d{4}|-?)-?",
-    two,
-    "-?",
-    two
-  ))
-  partial <- vector("list", 3)
-  components <- c("year", "month", "day")
-  names(partial) <- components
-  for (i in seq_along(components)) {
-    partial[[i]] <- partialdate[, i + 1]
-    partial[[i]] <- if_else(partial[[i]] %in% c("-", ""), NA_character_, partial[[i]])
+  # the `assert_date_imputation` function is stored in `derive_vars_dt_dtm_utils.R`
+  date_imputation <- assert_date_imputation(
+    highest_imputation = highest_imputation,
+    date_imputation = date_imputation
+  )
+
+  if (length(dtc) == 0) {
+    return(character(0))
   }
 
+  # Parse partials
+  partial <- parse_partial_date_time(dtc, is_datetime)
```

Comment:
```suggestion
  partial <- get_partialdatetime(dtc, create_datetime = FALSE)
```
See comment for `parse_partial_date_time()`.

Reply:
Done